### PR TITLE
Refactored serializations

### PR DESCRIPTION
**Pull request by DanielaIvanova**

### CODE REVIEW

File: apps/utils/lib/serialization.ex [hunk 1]
```diff
@@ -45,24 +46,298 @@ defmodule Utils.Serialization do
...
+    process_serialize(fields, type)
+  end
+
+  # this is the way the id record is represented in erlang
+  @spec id_to_record(binary(), id_type()) :: id()
+  def id_to_record(value, type)
+      when type in [:account, :oracle, :name, :commitment, :contract, :channel],
+      do: {:id, type, value}
+
+  defp process_tx_fields(tx, :spend_tx) do
```
Comment (cheezus1) [inline apps/utils/lib/serialization.ex]: Can you drop the second parameter and use the corresponding model structure as the first parameter? SpendTx in this case, for example. Also, match all the fields inside the parameter.

**Comment by d-velev** [comment]
This adds 400 lines to the serialization module and I'm not sure there's an added benefit. The list building in the high-level modules will be replaced with structure building and that doesn't really make development easier.

**Comment by thepiwo** [comment]
@d-velev but all the lines of code would be scattered around the high-level modules then?

**Comment by d-velev** [comment]
@thepiwo no, originally we had the list of transaction parameters ([account, nonce, fee. . .]) in the high level module, now we have them in the serialization module and we need to structure building as well, there's no reduction

**Comment by thepiwo** [comment]
@d-velev like this right? https://github.com/aeternity/aepp-sdk-elixir/blob/1b177d52bab11c4e58904776e68e4a1c6bd066da/apps/core/lib/contract.ex#L77

I do actually prefer having this separate for simplicity, are there other reasons not to separate as done in this PR? Whats your opinion @DanielaIvanova @Artur64 ?

**Comment by d-velev** [comment]
This way we have to use structures just to pass data around internally, they aren't a function parameter or output (to benefit the end user). Also the serialization module was cleaner as it did only serialization

**Comment by thepiwo** [comment]
I don't have a strong opinion here, I just feel that the serialization structure has nothing to do in the high level code, as you said it only affects serialization.

**Comment by thepiwo** [comment]
I think the best solution would be to do it as in this PR but put all the `process_tx_fields` in a separate file.

**Comment by Artur64** [comment]
@d-velev Your idea with serializations is good but it doesn't work for us for implementing minimum fee calculation, because, so, in order to calculate fee, we will need to know the byte size of the transaction and without it we wont be able to calculate it.

**Comment by d-velev** [comment]
@Artur64  Why couldn't you get the size of the transaction before?

**Comment by DanielaIvanova** [comment]
@d-velev The transaction should be serialized first and we have to get the byte size then.

**Comment by d-velev** [comment]
@DanielaIvanova It can be serialized first the old way as well

**Comment by DanielaIvanova** [comment]
@d-velev Yes, that is true but we have to implement more than 18 functions for calculating minimum fee for each type of transaction. Also if we need to serialize transaction afterwards we will have to implement again those functions.

**Comment by thepiwo** [comment]
Okay, lets stop discussion here and agree this is a good solution? @d-velev @cheezus1 

@DanielaIvanova @Artur64 could you maybe refactor so all the `process_tx_fields` are in a separate file to get a better overview?

**Comment by DanielaIvanova** [comment]
OK, we will move to a separate file.

**Pull request merged**